extend event logging for debugging purposes

rclee opened the pull request:
-adds new method to MMEEventLogger for writing events to a local file
-writes any event passing through sendTurnstile or pushEvent methods in MMEEventsManager

rclee committed:
extend event logging

-adds new method to MMEEventLogger for writing events to a local file
-writes any event passing through sendTurnstile or pushEvent methods in MMEEventsManager

boundsj commented on MapboxMobileEvents/MMEEventLogger.h:
```diff
@@ -10,6 +10,8 @@ NS_ASSUME_NONNULL_BEGIN
 + (void)setEnabled:(BOOL)enabled;
```
We could probably make `enabled` a property.

boundsj reviewed (commented):
Looking good! Please have a look at the comments wrt rationalizing how we use this class as a shared instance vs one off instances.

rclee commented on MapboxMobileEvents/MMEEventLogger.m:
```diff
@@ -18,4 +26,44 @@ + (void)setEnabled:(BOOL)enabled {
     _enabled = enabled;
 }
 
+- (void)writeEventToLocalDebugLog:(MMEEvent *)event {
```
I definitely like this approach better. Initially I had similar thoughts with combining the two methods. My only concern was potentially having too many logs written locally and that a majority of the debug logs will have the same ``MMEEventKeyLocalDebug`` key.

It seems like we'll still need to have other keys for at least ``flush`` and ``push`` and that would likely result in any ``MMEEventKeyLocalDebug`` key/value pairs being unused; unless that information is interesting in some way...

rclee committed:
Add MMEEventLogger singleton and event types

boundsj commented on MapboxMobileEvents/MMEEventsManager.m:
```diff
@@ -157,6 +157,12 @@ - (void)flush {
     
+    NSDictionary *flushEventAttributes = @{MMEEventKeyEvent: MMEEventTypeFlush,
+                                          MMEEventKeyCreated: [self.dateWrapper formattedDateStringForDate:[self.dateWrapper date]]};
+    MMEEvent *flushEvent = [MMEEvent pushEventWithAttributes:flushEventAttributes];
```
With the changes in this PR so far, I think this should actually be `flushEventWithAttributes:`. But let's make the change to use the unified `debugEventWithAttributes`. As a debug event, the event name will be `MMEEventTypeLocalDebug` (that is done for us by `debugEventWithAttributes:`) and we could introduce a "debug event type" that we put in the attributes. In this case, that could be a string constant called  `MMEDebugEventTypeFlush` that equals "flush".

boundsj commented on MapboxMobileEvents/MMEEventsManager.m:
```diff
@@ -375,7 +389,7 @@ - (void)pushDebugEventWithAttributes:(MMEMapboxEventAttributes *)attributes {
     [combinedAttributes setObject:self.uniqueIdentifer.rollingInstanceIdentifer forKey:@"instance"];
     MMEEvent *debugEvent = [MMEEvent debugEventWithAttributes:combinedAttributes];
-    [MMEEventLogger logEvent:debugEvent];
+    [MMEEventLogger.sharedLogger logEvent:debugEvent];
```
For all of the existing invocations of `pushDebugEventWithAttributes:` we should make sure their attributes have a `MMEDebugEventType` and `MMEDebugEventType{foo}` value that makes sense for the debug log entries they represent.

boundsj reviewed (changes requested):
Thanks for making the changes to allow for using the logger as a singleton. This helped illustrate for me some other changes I think we should make.

rclee committed:
Add DebugTypes to MMEConstants

-introduces MMEDebugEventTypes to MMEConstants
-adds debug types to pushDebugEventWithAttributes:

rclee committed:
Remove old methods

rclee committed:
Fix missing key

boundsj commented on MapboxMobileEvents/MMEEventLogger.h:
```diff
@@ -6,9 +6,13 @@ NS_ASSUME_NONNULL_BEGIN
+
+- (void)logEvent:(MMEEvent *)event;
+
+- (void)writeEventToLocalDebugLog:(MMEEvent *)event;
```
I guess we could remove this in the header file and make it an implementation detail.

rclee committed:
Remove writeEventToLocalDebugLog: from header
